src/Features/CSharp/Portable/ConvertLinq/CSharpConvertLinqQueryToForEachProvider.cs

```diff
@@ -0,0 +1,792 @@
+                    StringComparer.CurrentCulture);
+
+            private bool TryConvertIfInForEach(QueryExpressionSyntax source, ForEachStatementSyntax forEachStatement, out DocumentUpdateInfo documentUpdateInfo)
+            {
+                // before foreach(var x in from a in b select a)
+                var statement = forEachStatement.Statement;
+                StatementSyntax internalNodeMethod(ExpressionSyntax expression)
+                {
+                    // before 
+                    //  foreach (var a in from a in b where a > 5 select a) 
+                    //  { 
+                    //      dosomething(a); 
+                    //  }
+                    // after 
+                    //  foreach (var a in b)
+                    //  {
+                    //      if (a > 5)
+                    //      {
+                    //          dosomething(a); 
+                    //      }
+                    //  }
+                    if (expression is IdentifierNameSyntax identifier &&
+                        identifier.Identifier.ValueText == forEachStatement.Identifier.ValueText)
+                    {
+                        return statement.WithAdditionalAnnotations(Formatter.Annotation);
+                    }
+                    else
+                    {
+                        // before foreach(var x in from ... a) { dosomething(x); }
+                        // after 
+                        // foreach (var a in ...)
+                        // ...
+                        // {
+                        //      var x = a;
+                        //      dosomething(x); 
+                        //  }
+                        var declaration = SyntaxFactory.LocalDeclarationStatement(
+                            SyntaxFactory.VariableDeclaration(
+                                VarNameIdentifier,
+                                SyntaxFactory.SingletonSeparatedList(
+                                    SyntaxFactory.VariableDeclarator(
+                                        forEachStatement.Identifier,
+                                        argumentList: null,
+                                        SyntaxFactory.EqualsValueClause(expression)))));
+                        return AddToBlockTop(declaration, statement).WithAdditionalAnnotations(Formatter.Annotation);
+                    }
+                }
+
+                if (TryConvertQueryExpression(source, internalNodeMethod, out StatementSyntax newNode))
+                {
+                    documentUpdateInfo = new DocumentUpdateInfo(forEachStatement, newNode.WithAdditionalAnnotations(Simplifier.Annotation));
+                    return true;
+                }
+
+                documentUpdateInfo = default;
+                return false;
+            }
+
+            private bool TryConvertIfInReturnStatement(QueryExpressionSyntax source, ReturnStatementSyntax returnStatement, out DocumentUpdateInfo documentUpdateInfo)
```

Comment:
>TryConvertIfInReturnStatement [](start = 25, length = 29)

Are we verifying that the return statement is the last statement in function? Otherwise, I would expect to see ```yield break;``` added after the loop